GameDataParser/Parsers/MapEntityParser.cs

```diff
@@ -100,386 +68,345 @@ protected override List<MapEntityMetadata> Parse()
+            Entities.Sort((metadata1, metadata2) => metadata1.MapId.CompareTo(metadata2.MapId));
+            return Entities;
+        }
+
+        private void BuildMetadata(string xblock, IEnumerable<IMapEntity> mapEntities)
+        {
+            if (xblock.EndsWith("_cn") || xblock.EndsWith("_jp") || xblock.EndsWith("_kr"))
             {
-                Match isParsableField = Regex.Match(Path.GetFileNameWithoutExtension(entry.Name), @"^(\d{8})");
-                if (!isParsableField.Success)
-                {
-                    // TODO: Handle these later, if we need them. They're xblock files with some other names like
-                    //  character_test.xblock, login.xblock, 
-                    continue;
-                }
+                return;
+            }
 
-                string mapId = isParsableField.Groups[1].Value;
-                if (maps.ContainsKey(mapId))
-                {
-                    continue;
-                }
-                maps.Add(mapId, entry.Name);  // Only used to check if we've visited this node before.
+            Match isParsableField = Regex.Match(xblock, @"^(\d{8})");
+            if (!isParsableField.Success)
+            {
+                // TODO: Handle these later, if we need them. They're xblock files with some other names like
+                //  character_test.xblock, login.xblock, 
+                return;
+            }
 
-                MapEntityMetadata metadata = new MapEntityMetadata(int.Parse(mapId));
-                XmlDocument document = Resources.ExportedReader.GetXmlDocument(entry);
+            string mapId = isParsableField.Groups[1].Value;
+            if (Maps.ContainsKey(mapId))
+            {
+                return;
+            }
 
-                foreach (XmlNode node in document.SelectNodes("/game/entitySet/entity"))
-                {
-                    string modelName = node.Attributes["modelName"].Value;  // Always maps to a .flat fileName for additional supplemented data
-                    string name = node.Attributes["name"].Value;
+            Maps.Add(mapId, xblock); // Only used to check if we've visited this node before.
 
-                    if (modelName == "MS2Bounding")
-                    {
-                        XmlNode blockCoord = node.SelectSingleNode("property[@name='Position']");
-                        CoordS boundingBox = CoordS.Parse(blockCoord?.FirstChild.Attributes["value"].Value ?? "0, 0, 0");
-                        if (name.EndsWith("0") && metadata.BoundingBox0.Equals(CoordS.From(0, 0, 0)))
+            // TODO: metadata should be keyed on xblock name, not mapId
+            MapEntityMetadata metadata = new MapEntityMetadata(int.Parse(mapId));
+
+            foreach (IMapEntity entity in mapEntities)
+            {
+                switch (entity)
+                {
+                    case IMS2Bounding bounding:
+                        if (bounding.EntityName.EndsWith("0"))
```

Comment:
```suggestion
                        if (bounding.EntityName.EndsWith("0") && metadata.BoundingBox0.Equals(CoordS.From(0, 0, 0))
```
this code is needed even though it looks weird. Accept only the first coord in the xmls, since some maps have more than one bounding box.

Reply:
was it in the old parser?

Reply:
Yes it was

Reply:
have you tested whether it works as intended?